### PR TITLE
Private events app

### PR DESCRIPTION
**Pull request by MkrtichSargsyan**
In this project, I built a private events app. It allows users to create events and then manage user signups. A user can create events, can attend many events. An event can be attended by many users

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  def create
+    @event = current_user.created_events.build(event_params)
+    @event.save
+    if @event.save
+      redirect_to @event, notice: 'Your event was created'
+    else
+      render :new
+    end
+  end
```
Comment (Dandush03) [inline app/controllers/events_controller.rb]: when we create an event, we need to validate it first and then create it, calling twice the save action is not a good practice, and can cause some double creation if there isn't proper validation to it, so we can remove the first call, or do it the following way (best practice)
```ruby
 def create
    @event = current_user.created_events.build(event_params)
    if @event.valid?
      @event.save
      redirect_to @event, notice: 'Your event was created'
    else
      render :new
    end
  end
```

File: config/routes.rb [hunk 2]
```diff
@@ -1,4 +1,18 @@
...
+  resources :users
+  resources :events
+
+  resources :attendances
+
+  resources :attendances do
+    match '/join', to: 'attendances#join', via: :post, on: :collection
+  end
+
+  root 'events#index'
```
Comment (Dandush03) [inline config/routes.rb]: please use proper flags for each resources, for example:
```ruby
sources :events, only: %i[index]
```

**Review by Dandush03 — CHANGES_REQUESTED** [review]
# Status: :red_circle: Require Changes :red_circle: 
### Great work so far, but we most fix some issue in order to get approved :wink: 

- [ ] **Missing Requirements:**
  - [ ] **EventActions** When a user is logged in Event#index Lists all the existing events separated into Past and Upcoming categories
  - [ ] **UserActions** When a user is logged in Visit a User's show page lists all the events the user has created
  - [ ] **UserActions** When a user is logged in Visit a User's show page lists all the events where the user is an attendee
  - [ ] **User** can retrieve past events the user has attended/created
  - [ ] **User** can retrieve future events the user has created / will attend
- [ ] **Code Issues:**
  - [ ] **Event Controller:** [here you can see the issue](https://github.com/MkrtichSargsyan/private-events/pull/1#discussion_r538892819)
  - [ ] **Routes:** [here you can see the issue](https://github.com/MkrtichSargsyan/private-events/pull/1#discussion_r538896134)

### OPTIONAL
- [ ] Change your event column name `desc` to `content` or `description` since `desc` is an SQL function
# Happy Coding :+1: :tophat:

**Review by lekegitrepo — APPROVED** [review]
## Status: Approved :heavy_check_mark:
### Hi @MkrtichSargsyan :100: :1st_place_medal:
#### Well done :muscle: you have done an incredible job, the user interface looks great. This project is approved.

### Optional:
- **_Implement user authentication tests: integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))_**

- **_Implement event managment features tests: integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))_**

#### _Reviewed by Temitayo Adeleke_

**Pull request merged**